safe_transaction_service/history/services/transaction_service.py

```diff
@@ -282,57 +288,87 @@ def get_all_tx_identifiers(
 
     def get_all_txs_from_identifiers(
         self, safe_address: str, ids_to_search: Sequence[str]
-    ) -> List[Union[EthereumTx, MultisigTransaction, ModuleTransaction]]:
+    ) -> List[AnySafeTransaction]:
         """
         Now that we know how to paginate, we retrieve the real transactions
 
         :param safe_address:
         :param ids_to_search: `SafeTxHash` for MultisigTransactions, `txHash` for other transactions
         :return:
         """
-        cached_txs = {
-            id_to_search: cached_tx
-            for id_to_search, cached_tx in zip(
+
+        logger.debug(
+            "Safe=%s Getting %d txs from identifiers", safe_address, len(ids_to_search)
+        )
+        ids_with_cached_txs = {
+            id_to_search: cached_txs
+            for id_to_search, cached_txs in zip(
                 ids_to_search,
                 self.get_txs_from_cache(safe_address, ids_to_search),
             )
-            if cached_tx
+            if cached_txs
         }
-        id_not_cached = [
+        logger.debug(
+            "Safe=%s Got %d cached txs from identifiers",
+            safe_address,
+            len(ids_with_cached_txs),
+        )
+        ids_not_cached = [
             hash_to_search
             for hash_to_search in ids_to_search
-            if hash_to_search not in cached_txs
+            if hash_to_search not in ids_with_cached_txs
         ]
-        id_with_multisig_txs: Dict[HexStr, List[MultisigTransaction]] = {
+        logger.debug(
+            "Safe=%s %d not cached txs from identifiers",
+            safe_address,
+            len(ids_not_cached),
+        )
+        ids_with_multisig_txs: Dict[HexStr, List[MultisigTransaction]] = {
             multisig_tx.safe_tx_hash: [multisig_tx]
             for multisig_tx in MultisigTransaction.objects.filter(
-                safe=safe_address, safe_tx_hash__in=id_not_cached
+                safe=safe_address, safe_tx_hash__in=ids_not_cached
             )
             .with_confirmations_required()
             .prefetch_related("confirmations")
             .select_related("ethereum_tx__block")
             .order_by("-nonce", "-created")
         }
+        logger.debug(
+            "Safe=%s Got %d Multisig txs from identifiers",
```

Comment:
Maybe better indicate that we are getting from database filtering by identifiers. The same for the other logs.

Reply:
I'm not understanding this, we are already stating that we are getting multisig tx from identifiers

Reply:
I haven't strong opinion about it, but reading the logs without context I'd prefer indicate that the data is coming from database, because the other way is like the identifiers list include the transactions information and it is not true.

Reply:
But I mean, Multisig Tx from identifiers can only be fetched from database (previously we logged the cache ones) so I don't think we have to be explicit about it